lib/namma-dsl/src/NammaDSL/Lib/Interpreter.hs

```diff
@@ -0,0 +1,75 @@
+module NammaDSL.Lib.Interpreter where
+
+import Control.Monad.Writer hiding (Writer)
+import qualified Data.List as L
+import GHC.IO (unsafePerformIO)
+import Kernel.Prelude
+import qualified Language.Haskell.TH as TH
+import qualified Language.Haskell.TH.Ppr as Ppr
+import qualified Language.Haskell.TH.PprLib as Ppr
+import qualified NammaDSL.Lib.MapName as MapName
+import NammaDSL.Lib.Types
+import qualified Text.PrettyPrint as Pretty
+
+-- interpreter :: String -> Writer CodeUnit -> String
+-- interpreter moduleName unitW = do
+interpreter :: [String] -> Writer CodeUnit -> String
+interpreter imports unitW = do
+  let codeUnits = unsafePerformIO . TH.runQ . execWriterT $ unitW -- FIXME remove unsafe
+  let codeTree = foldMap mkCodeTree codeUnits
+  interpret imports codeTree
+
+mkCodeTree :: CodeUnit -> CodeTree
+mkCodeTree (CodeDec codeDecs) = mempty{codeDecs = codeDecs}
+-- mkCodeTree (CodeImport codeImport) = mempty{codeImports = [codeImport]}
+-- mkCodeTree (CodeExtension codeExtension) = mempty{codeExtensions = [codeExtension]}
+mkCodeTree (CodeSplice codeSplice') = mempty{codeSplices = [codeSplice']}
+
+interpret :: [String] -> CodeTree -> String
+interpret imports CodeTree {..} = do
+  -- interpret :: String -> CodeTree -> String
+  -- interpret moduleName CodeTree {..} = do
+  -- L.intercalate "\n" (interpretExtension <$> codeExtensions)
```

Comment:
comments will be removed when all code will be changed to TH, because now output string constructed in other way